src/gator/core/models/timetable.py

```diff
@@ -13,131 +13,116 @@
+        kwargs['subsession'] = subsession
         super().__init__(*args, **kwargs)
 
     @property
     def code(self) -> str:
-        """The canonical representation of this session.
-
-        Returns:
-            A five-length string where the first four characters denote the
-            session year, and the last character denotes the session season,
-            9 for fall, 1 for winter, and 5 for summer.
+        """Return the canonical representation of the session.
 
-        Examples:
-            >>> Session(2020, 'fall').code
-            '20209'
-            >>> Session(2019, 'winter').code
-            '20191'
-            >>> Session(1966, 'summer').code
-            '19665'
-        """
-        return f'{str(self.year).zfill(4)}{self._SEASON_MAP[self.season]}'
-
-    @property
-    def human_str(self) -> str:
-        """A human-readable representation of this session.
+        The canonical representation of a session is a string whose format depends on the
+        season of the session.
 
         Examples:
-            >>> Session(2020, 'fall').human_str
-            'Fall 2020'
-            >>> Session(2019, 'winter').human_str
-            'Winter 2019'
-            >>> Session(1966, 'summer').human_str
-            'Summer 1966'
+            >>> Session(2022, 'regular', 'first').code
+            '20229'
+            >>> Session(2022, 'regular', 'second').code
+            '20231'
+            >>> Session(2022, 'regular', 'whole').code
+            '20229-20231'
+            >>> Session(2023, 'summer', 'first').code
+            '20235F'
+            >>> Session(2023, 'summer', 'second').code
+            '20235S'
+            >>> Session(2023, 'summer', 'whole').code
+            '20235'
         """
-        return f'{self.season.capitalize()} {self.year}'
+        assert self.season in ['regular', 'summer']
+
+        if self.season == 'regular':
+            if self.subsession == 'first':
+                return f'{self.year}9'
+            elif self.subsession == 'second':
+                return f'{self.year}1'
+            else:
+                return f'{self.year}9-{self.year}1'
+        else:
+            if self.subsession == 'first':
+                return f'{self.year}5F'
+            elif self.subsession == 'second':
+                return f'{self.year}5S'
+            else:
+                return f'{self.year}5'
 
     def __str__(self) -> str:
-        """Return a string representation of this session.
-
-        The output string is the session code. See `code` for more
-        information.
-        """
+        """Return a string representation of the session."""
         return self.code
 
     @classmethod
-    def parse(cls, session_code: Union[str, int]) -> 'Session':
-        """Return an instance Session representing the given session code.
-
-        Raise a ValueError if the session code is not formatted properly.
+    def from_code(cls, code: str) -> 'Session':
```

Comment:
nit: is there any reason why we renamed this method from `parse`?

Reply:
I will alias this to maintain the interface.